L/LLVM/common.jl

```diff
@@ -395,7 +398,12 @@ function configure_build(ARGS, version; experimental_platforms=false, assert=fal
     end
     if version >= v"12"
         push!(products, LibraryProduct("libclang-cpp", :libclang_cpp, dont_dlopen=true))
-        push!(products, ExecutableProduct("lld", :lld, "bin"))
+        push!(products, ExecutableProduct("lld", :lld, "tools"))
```

Comment:
```suggestion
        push!(products, ExecutableProduct("lld", :lld, "bin"))
```

@vchuravy does this ^^ seem correct? Or do we want to move `lld` to `tools/` even earlier?

Reply:
https://github.com/llvm/llvm-project/commit/352ac408d4ff7705529506bf7dbca4de6b32cef4 but for LLD?

Reply:
Ok latest commit does this for LLVM 12.0.1. Want this for all 12+ versions?

Reply:
Jup, but we can do that in stages. We probably want to add them to `JuliaLang/llvm-project`